### PR TITLE
[wip] test: workaround for nested describe skips

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Corrects some tests that we were incorrectly marking as passed when they should be explicitly skipped instead.

cc @alexeykuzmin 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: spec/api-auto-updater-spec.js [hunk 1]
```diff
@@ -3,21 +3,15 @@ const { ipcRenderer } = require('electron')
...
+    if (process.mas) this.skip()
   })
 
   describe('checkForUpdates', function () {
     it('emits an error on Windows when called the feed URL is not set', function (done) {
       if (process.platform !== 'win32') {
-        // FIXME(alexeykuzmin): Skip the test.
-        // this.skip()
-        return done()
+        this.test.parent.pending = true
```
Comment (MarshallOfSound) [inline spec/api-auto-updater-spec.js]: I'm not a fan of hitting this internal mocha API.

Might be slightly biased but I prefer the approach I started taking in #15782 with declarative `platformDescribe` and `platformIt` methods.

File: spec/api-auto-updater-spec.js [hunk 2]
```diff
@@ -3,21 +3,15 @@ const { ipcRenderer } = require('electron')
...
+        this.skip()
```
Comment (alexeykuzmin) [inline spec/api-auto-updater-spec.js]: `done` still has to be called after `.skip()`.

File: spec/api-auto-updater-spec.js [hunk 3]
```diff
@@ -55,9 +48,7 @@ describe('autoUpdater module', function () {
       }
 
       before(function () {
-        if (process.platform !== 'win32' && process.platform !== 'darwin') {
-          this.skip()
-        }
+        if (process.platform === 'linux') this.skip()
```
Comment (alexeykuzmin) [inline spec/api-auto-updater-spec.js]: 👍

File: spec/api-auto-updater-spec.js [hunk 4]
```diff
@@ -125,9 +114,8 @@ describe('autoUpdater module', function () {
   describe('quitAndInstall', () => {
     it('emits an error on Windows when no update is available', function (done) {
       if (process.platform !== 'win32') {
-        // FIXME(alexeykuzmin): Skip the test.
-        // this.skip()
-        return done()
+        this.test.parent.pending = true
+        this.skip()
```
Comment (alexeykuzmin) [inline spec/api-auto-updater-spec.js]: Call `done()` and `return`

**Comment by alexeykuzmin** [comment]
I like half of the changes, but agree to Sam that usage of the internal API isn't great.
It can easily break when we update to a newer version of Mocha.

**Comment by codebytere** [comment]
Yeah, that's fair. Let's shelve this one for now and see if there's a better approach we can potentially take!

**Pull request closed without merging**